corehq/apps/sms/models.py

```diff
@@ -2433,5 +2436,17 @@ class KeywordAction(models.Model):
     # This can be None in which case there is no separator (i.e., "report a100 b200")
     named_args_separator = models.CharField(max_length=126, null=True)
 
+    def save(self, *args, **kwargs):
+        if self.recipient == self.RECIPIENT_USER_GROUP and not self.recipient_id:
+            raise self.InvalidModelStateException("Expected a value for recipient_id")
+
+        if self.action == self.ACTION_SMS and not self.message_content:
```

Comment:
i didn't think so, it would just be ignored for the rest of the actions so it's not a big deal